src/ZulipMobile.js

```diff
@@ -23,19 +25,21 @@ console.disableYellowBox = true; // eslint-disable-line
 export default (): React$Node => (
   <CompatibilityChecker>
     <StoreProvider>
-      <AppEventHandlers>
-        <AppDataFetcher>
-          <TranslationProvider>
-            <ThemeProvider>
-              <InitialNavigationDispatcher>
-                <BackNavigationHandler>
-                  <AppContainer ref={NavigationService.appContainerRef} />
-                </BackNavigationHandler>
-              </InitialNavigationDispatcher>
-            </ThemeProvider>
-          </TranslationProvider>
-        </AppDataFetcher>
-      </AppEventHandlers>
+      <HideIfNotHydrated PlaceholderComponent={LoadingScreen}>
+        <AppEventHandlers>
```

Comment:
Oh, right. Hmm. So, during rehydration, the default values will be used. A particular problem is that the safe-area insets default to zero. (That doesn't matter when the orientation is landscape, since the status bar isn't shown and the safe-area insets are ignored. But the default value for the orientation is portrait.)

We currently use [`react-native-safe-area`](https://www.npmjs.com/package/react-native-safe-area) to asynchronously get the safe-area insets on startup and put them in Redux.

I wonder about using the much-more-popular [`react-native-safe-area-context`](https://www.npmjs.com/package/react-native-safe-area-context) (already in our dependencies) instead. Once you've set up a `SafeAreaProvider `, it looks like you can subscribe any component to the safe-area insets using a Hook, a HOC, or a Context consumer. This seems a more natural approach than depending on a dispatch to Redux, maybe? And it looks like `react-native-safe-area` doesn't have something like this.

Reply:
> I wonder about using the much-more-popular [`react-native-safe-area-context`](https://www.npmjs.com/package/react-native-safe-area-context) (already in our dependencies) instead.

Just opened https://github.com/zulip/zulip-mobile/pull/4315 for this.